### PR TITLE
Handle app config errors

### PR DESCRIPTION
**Pull request by benrudolph**
@wpride @snopoke this'll handle handle app config errors and display the message. i don't think this'll yet handle xpath errors, but all we need to do is catch those and re wrap as an ApplicationConfigError and that'll stop the emails

screen shot 2016-08-29 at 10 00 53 pm

I know it's ugly at the moment, going to work on fixing later

cross: https://github.com/dimagi/commcare-hq/pull/13038

### CODE REVIEW

File: src/main/java/install/FormplayerConfigEngine.java [hunk 1]
```diff
@@ -148,6 +154,18 @@ public void initFromLocalFileResource(String resource) throws InstallCancelledEx
         init(reference);
     }
 
+    private void handleInstallError(InputStream errorStream) {
+        StringWriter writer = new StringWriter();
+        try {
+            IOUtils.copy(errorStream, writer, "utf-8");
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (snopoke) [inline src/main/java/install/FormplayerConfigEngine.java]: This should rather log the error.

**Comment by snopoke** [comment]
:+1:

File: src/main/java/install/FormplayerConfigEngine.java (hunk 1 above)
Comment (benrudolph) [inline src/main/java/install/FormplayerConfigEngine.java]: ah good to know, thanks

Comment (benrudolph) [inline src/main/java/install/FormplayerConfigEngine.java]: https://github.com/dimagi/formplayer/pull/102/commits/793a4107629b237db12fc09f05dacbbb8faab730

**Comment by wpride** [comment]
@benrudolph probably XPathException (and its sub classes) belong in the new handler too

**Comment by wpride** [comment]
but 👍 on ✅

**Comment by benrudolph** [comment]
right on, updated to catch base class

**Pull request merged**